{
  "cve_id": "CVE-2014-3589",
  "github_data": {
    "repository": "python-pillow/Pillow",
    "fix_commit": "205e056f8f9b06ed7b925cf8aa0874bc4aaf8a7d",
    "related_commits": [
      "205e056f8f9b06ed7b925cf8aa0874bc4aaf8a7d",
      "205e056f8f9b06ed7b925cf8aa0874bc4aaf8a7d"
    ],
    "patch_url": "https://github.com/python-pillow/Pillow/commit/205e056f8f9b06ed7b925cf8aa0874bc4aaf8a7d.patch",
    "fix_commit_details": {
      "sha": "205e056f8f9b06ed7b925cf8aa0874bc4aaf8a7d",
      "commit_date": "2014-08-06T23:42:43Z",
      "author": {
        "login": "wiredfool",
        "type": "User",
        "stats": {
          "total_commits": 2729,
          "average_weekly_commits": 3.60978835978836,
          "total_additions": 0,
          "total_deletions": 0,
          "weeks_active": 217
        }
      },
      "commit_message": {
        "title": "Icns DOS fix --  CVE-2014-3589",
        "length": 81,
        "has_description": true,
        "references_issue": true
      },
      "stats": {
        "total": 12,
        "additions": 12,
        "deletions": 0
      },
      "files": [
        {
          "filename": "PIL/IcnsImagePlugin.py",
          "status": "modified",
          "additions": 2,
          "deletions": 0,
          "patch": "@@ -179,6 +179,8 @@ def __init__(self, fobj):\n         i = HEADERSIZE\n         while i < filesize:\n             sig, blocksize = nextheader(fobj)\n+            if blocksize <= 0:\n+                raise SyntaxError('invalid block header')\n             i += HEADERSIZE\n             blocksize -= HEADERSIZE\n             dct[sig] = (i, blocksize)"
        },
        {
          "filename": "Tests/check_icns_dos.py",
          "status": "added",
          "additions": 10,
          "deletions": 0,
          "patch": "@@ -0,0 +1,10 @@\n+# Tests potential DOS of IcnsImagePlugin with 0 length block.\n+# Run from anywhere that PIL is importable. \n+\n+from PIL import Image\n+from io import BytesIO\n+\n+if bytes is str:\n+    Image.open(BytesIO(bytes('icns\\x00\\x00\\x00\\x10hang\\x00\\x00\\x00\\x00')))\n+else:\n+    Image.open(BytesIO(bytes('icns\\x00\\x00\\x00\\x10hang\\x00\\x00\\x00\\x00', 'latin-1')))"
        }
      ],
      "file_patterns": {
        "security_files": 0,
        "config_files": 0,
        "dependency_files": 0,
        "test_files": 1,
        "unique_directories": 2,
        "max_directory_depth": 1
      },
      "context": {
        "surrounding_commits": [
          {
            "sha": "84d58a0282df5ec081502f99f2c363b79c513ee4",
            "date": "2025-01-14T15:04:39Z",
            "author_login": "hugovk"
          },
          {
            "sha": "bdfd881c6df81c10697f68f204582aa16a451d1e",
            "date": "2025-01-14T14:24:41Z",
            "author_login": "hugovk"
          },
          {
            "sha": "cf438c53eed627e62baff752cb874aad3bcf63d5",
            "date": "2025-01-14T10:04:08Z",
            "author_login": "radarhere"
          },
          {
            "sha": "0f2c554c698266ec0ba464f21a616cd0eda9a7fb",
            "date": "2025-01-14T10:03:20Z",
            "author_login": "radarhere"
          },
          {
            "sha": "f521a4be7db5fb1b6060d1dc48e69ee3bdec25ab",
            "date": "2025-01-13T19:21:06Z",
            "author_login": "hugovk"
          }
        ]
      }
    }
  },
  "vulnerability_details": {
    "cvss_score": null,
    "cvss_vector": null,
    "cwe_id": "CWE-20",
    "description": "PIL/IcnsImagePlugin.py in Python Imaging Library (PIL) and Pillow before 2.3.2 and 2.5.x before 2.5.2 allows remote attackers to cause a denial of service via a crafted block size.",
    "attack_vector": null,
    "attack_complexity": null
  },
  "temporal_data": {
    "published_date": "2014-08-25T14:55:06.643",
    "last_modified": "2024-11-21T02:08:27.560",
    "fix_date": "2014-08-06T23:42:43Z"
  },
  "references": [
    {
      "url": "http://lists.opensuse.org/opensuse-updates/2015-04/msg00056.html",
      "source": "secalert@redhat.com",
      "tags": []
    },
    {
      "url": "http://secunia.com/advisories/59825",
      "source": "secalert@redhat.com",
      "tags": []
    },
    {
      "url": "http://www.debian.org/security/2014/dsa-3009",
      "source": "secalert@redhat.com",
      "tags": []
    },
    {
      "url": "https://github.com/python-pillow/Pillow/commit/205e056f8f9b06ed7b925cf8aa0874bc4aaf8a7d",
      "source": "secalert@redhat.com",
      "tags": [
        "Patch"
      ]
    },
    {
      "url": "https://pypi.python.org/pypi/Pillow/2.3.2",
      "source": "secalert@redhat.com",
      "tags": [
        "Patch"
      ]
    },
    {
      "url": "https://pypi.python.org/pypi/Pillow/2.5.2",
      "source": "secalert@redhat.com",
      "tags": [
        "Patch"
      ]
    },
    {
      "url": "http://lists.opensuse.org/opensuse-updates/2015-04/msg00056.html",
      "source": "af854a3a-2127-422b-91ae-364da2661108",
      "tags": []
    },
    {
      "url": "http://secunia.com/advisories/59825",
      "source": "af854a3a-2127-422b-91ae-364da2661108",
      "tags": []
    },
    {
      "url": "http://www.debian.org/security/2014/dsa-3009",
      "source": "af854a3a-2127-422b-91ae-364da2661108",
      "tags": []
    },
    {
      "url": "https://github.com/python-pillow/Pillow/commit/205e056f8f9b06ed7b925cf8aa0874bc4aaf8a7d",
      "source": "af854a3a-2127-422b-91ae-364da2661108",
      "tags": [
        "Patch"
      ]
    },
    {
      "url": "https://pypi.python.org/pypi/Pillow/2.3.2",
      "source": "af854a3a-2127-422b-91ae-364da2661108",
      "tags": [
        "Patch"
      ]
    },
    {
      "url": "https://pypi.python.org/pypi/Pillow/2.5.2",
      "source": "af854a3a-2127-422b-91ae-364da2661108",
      "tags": [
        "Patch"
      ]
    }
  ],
  "collection_metadata": {
    "collected_at": "2025-01-11T22:58:25.524088",
    "processing_status": "raw"
  },
  "repository_context": {
    "name": "Pillow",
    "owner": "python-pillow",
    "created_at": "2012-07-24T21:38:39Z",
    "updated_at": "2025-01-14T09:28:37Z",
    "pushed_at": "2025-01-13T19:21:06Z",
    "size": 206276,
    "stars": 12469,
    "forks": 2245,
    "open_issues": 126,
    "watchers": 12469,
    "has_security_policy": false,
    "default_branch": "main",
    "protected_branches": [],
    "languages": {
      "Python": 2250345,
      "C": 1274070,
      "HTML": 16910,
      "PostScript": 11949,
      "Shell": 8022,
      "Makefile": 3825,
      "CMake": 3740,
      "Gnuplot": 1105,
      "Batchfile": 166
    },
    "commit_activity": {
      "total_commits_last_year": 2117,
      "avg_commits_per_week": 40.71153846153846,
      "days_active_last_year": 333
    },
    "security_features": {
      "has_security_policy": false,
      "has_protected_branches": false,
      "has_wiki": true,
      "has_issues": true,
      "allow_forking": true,
      "is_template": false,
      "license": "other"
    },
    "collected_at": "2025-01-14T13:39:48.765812"
  }
}